### ISSUE TITLE
iOS: Blurry src image when using PoiMarker

### ISSUE THREAD

**Issue by vedranivicatos** [mikes222/mapsforge_flutter#55]
#### Flutter version:
Flutter 3.0.0 • channel stable • https://github.com/flutter/flutter.git
Framework • revision ee4e09cce0 (5 weeks ago) • 2022-05-09 16:45:18 -0700
Engine • revision d1b9a6938a
Tools • Dart 2.17.0 • DevTools 2.12.2

#### Plugin version: 
**2.0.2** using:
```
  mapsforge_flutter:
    git:
      url: https://github.com/mikes222/mapsforge_flutter
      path: mapsforge_flutter
```

#### Device:
Apple iPhone 11, physical device, iOS 15.5

### **Description**
When using PoiMarker class to add a maker with an svg resource, the marker gets rendered but is quite blurry:
![IMG_0001](https://user-images.githubusercontent.com/106805181/173310201-4ddb1fa3-958c-4cbb-8145-02f850b3c413.PNG)

Here are the original svg files:
<img src="https://user-images.githubusercontent.com/106805181/173310433-fe991960-aa9c-4db4-9db8-8926da664166.svg" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/106805181/173310455-74024e39-3366-4c7b-9ce7-0c4709726dd4.svg" alt="drawing" width="300"/>
#### To reproduce

- Create ViewModel:
```
  ViewModel _createViewModel() {
    // in this demo we use the markers only for offline databases.
    ViewModel viewModel = ViewModel(
      displayModel: _displayModel,
      // contextMenuBuilder: DefaultContextMenuBuilder(),
    );
    viewModel.addOverlay(ZoomOverlay(viewModel));
    viewModel.addOverlay(DistanceOverlay(viewModel));
    //viewModel.addOverlay(DemoOverlay(viewModel: viewModel));

    // set default position
    viewModel.setMapViewPosition(_sharedPreferences!.getDouble(gpsLastLatitude)!,
        _sharedPreferences!.getDouble(gpsLastLongitude)!);
    viewModel.setZoomLevel(16);
    return viewModel;
  }
```
- Create MapModel:
```
  Future<MapModel> _createMapModel(String mapFilePath) async {
    MapFile mapFile = await MapFile.from(mapFilePath, null, null);
    /// For the offline-maps we need a cache for all the tiny symbols in the map
    RenderThemeBuilder renderThemeBuilder = RenderThemeBuilder();
    String content = await rootBundle.loadString("assets/render_themes/defaultrender.xml");
    renderThemeBuilder.parseXml(_displayModel, content);
    RenderTheme renderTheme = renderThemeBuilder.build();
    MapDataStoreRenderer jobRenderer = MapDataStoreRenderer(mapFile, renderTheme, _symbolCache, true);
    // final TileBitmapCache bitmapCache = await FileTileBitmapCache.create(jobRenderer.getRenderKey());

    MapModel mapModel = MapModel(
      displayModel: _displayModel,
      renderer: jobRenderer,
      symbolCache: _symbolCache
      // tileBitmapCache: bitmapCache,
    );

    // Add initial my location marker if in prefs
    if(_sharedPreferences?.getDouble(gpsLastLatitude) != null && _sharedPreferences?.getDouble(gpsLastLongitude) != null) {
      addMarker(
          _userSingleton.mcDataId,
          _sharedPreferences!.getDouble(gpsLastLatitude)!,
          _sharedPreferences!.getDouble(gpsLastLongitude)!
      );
    }
    mapModel.markerDataStores.add(_markerDataStore);
    return mapModel;
  }
```
- Create FlutterMapView:
```
  Future<FlutterMapView> _createFlutterMapView(String mapFilePath) async {
    _viewModel = _createViewModel();
    _mapModel = await _createMapModel(mapFilePath);
    return FlutterMapView(
        mapModel: _mapModel!,
        viewModel: _viewModel!,
    );
  }
```
- Add marker:
```
  void addMarker(String senderUri, double latitude, double longitude){
    bool isMyLocation = false;
    if(_markers.containsKey(senderUri)){
      _markerDataStore.removeMarker(_markers[senderUri]!);
    }
    _markers[senderUri] = PoiMarker(
        item: senderUri,
        src: isMyLocation ? "assets/images/my_gps_blue_circle.svg"
        : "assets/images/my_gps_green_circle.svg",
        latLong: LatLong(
            latitude,
            longitude
        ),
        width: 24,
        height: 24,
        markerCaption: isMyLocation ? null : MarkerCaption(
            text: senderUri,
            strokeColor: 0xff000000,
            fillColor: 0xff000000,
            captionOffsetY: -12,
            captionOffsetX: 55,
            strokeWidth: 0.0,
            fontSize: 20
        )
    )..initResources(_symbolCache);
    _markerDataStore.addMarker(_markers[senderUri]!);
  }
```

**Comment by mikes222** [comment]
Every graphical item in your sample image is blurry. Also the blue or violet lines. In order to explain that we need to talk about logical vs physical pixels. Flutter works with logical pixels which is great because for example 800px are roughly the same size at all devices so the developer does not need to deal with pixel resolutions. 

On the other hand a physical device often has many more pixels, for example 1920*1080, which leads to a factor of more than 2 in our example. Since we are rendering the tiles and markers in logical pixels you see the images as blurry as they are (they are scaled by the resolution factor).

Solution is that mapsforge should be aware of this discrepancy and render everything for physical pixel sizes (and make sure it is scaled accordingly so that the size of the map at the screen does not change compared to what we have now). 

- We need to find out this resolution factor (I know there is a widget for it)
- We need to make sure that mapsforge can work with larger tiles (so that we do not need to create that many tiles in order to keep performance problems to a minimum). 
- We need to scale the contents of the tiles to the physical/logical resolution while rendering it. 
- We need to make sure the text, symbols and markers are also scaled accordingly.
- We need to make sure that online tiles are still working

In other words: This is a huge task. You are invited to contribute (maybe for the first and second part, this seems rather easy). I doubt that I can adapt this by myself in the next weeks.

**Comment by vedranivicatos** [comment]
> * We need to find out this resolution factor (I know there is a widget for it)

I am not sure whether this is the widget but `MediaQuery.of(context).devicePixelRatio` returns 2.0 as ratio. Using this value as a deviceScaleFactor of the DisplayModel `DisplayModel _displayModel = DisplayModel(deviceScaleFactor: 2.0);` I get somewhat different rendering. This is the screenshot of the same area with scale factor 2.0:
![IMG_B6FC998122BE-1](https://user-images.githubusercontent.com/106805181/174574785-38b2bc7c-473d-4192-b3c4-00abe0e103d9.jpeg)

If I zoom out there is quite a difference in street names, symbols, icons etc. I will try to experiment some more with deviceScaleFactor/userScaleFactor/fontScaleFactor and render themes to see if I can hit the sweet spot.

**Comment by mikes222** [comment]
I am about to prepare a version which supports better tile-quality in the next days. There are still a couple of glitches to solve before I can commit.

**Comment by mikes222** [comment]
I have checked in now the first version to support better image quality. If you want to test it, change the displayModel and set ```DisplayModel(deviceScaleFactor: 2);```. I have not yet tested it with other factors. Still need to test some markers (circleMarker is already working). 
Changes made:
- Image of the tiles are improved
- pinch2zoom also affects markers
- more gesture detections (click, double-click, long-click, hold-then-move)

Feedback and error reports are welcome.

**Comment by vedranivicatos** [comment]
Great news, looking forward to test it out :D
